example-llm-workflows/README.md

```diff
@@ -0,0 +1,483 @@
+
+### Idempotent Replay
+
+Typically when relaying questions and prompts to LLM applications, you'll get varied responses and answers even when you don't change the input text.
+
+When a Homestar node receives a workflow containing a task that has already been executed, either with its index (key) and output (value) stored locally (most optimal) or retrieved from a shared DHT (Distributed Hash Table) over the network. In that case, it can replay that task without re-execution or having to fetch the Wasm module from its source. This replayability is akin to [durable function/execution][durable-fn] frameworks like [Cadence][cadence] and [Temporal][temporal] but happens over a decentralized mesh instead of a centralized infrastructure.
+
+So, what does that mean for us? If we provide the same arguments to our Guest Wasm functions, Homestar will return cached responses instantly, which saves us both time and resources, as waiting for large prompt responses when using models, even *smaller* ones like LLaMA 3 trained with 8 billion parameters, can be somewhat latent on devices with limited GPU and memory (like my personal computer).
+
+Is there a way to always provide *new* hallucinations from the Host model? Yes, there is! We can use the `--debug` flag when we start EveryCli to force re-execution of the tasks in our workflow.
+
+```console
+./every-cli/cli.js dev ../example-llm-workflows/llm.wasm --debug
+```
+
+## Foundations for Privacy
+
+The learning goals of this project were to experiment with working with LLMs locally on hosts where the training data and tuning of a model remains private and only derived information from prompt-based AI chains can be shared with other users/peers for consumption, allowing for working with AI computation in ways not tied to any specific vendor or large cloud provider. Essentially, this showcases a positive avenue for decentralized, user-controlled AI-oriented computation that's everything that [IEEE Spectrum's Open-Source AI Is Uniquely Dangerous][ieee] isn't.
+
+Localized, open-source LLM models for home and on-prem use cases are growing in popularity, even leading to the creation of a [LocalLLaMA reddit][reddit-post] community! We've seen how [GDPR][gdpr] has increased the need for companies to be more careful around [PII management and data privacy isolation across regions and region-compliance laws][so-privacy].
+
+Self-hosted, privately managed model deployments hit on many of the privacy and security modules taught in our course. Incorporating ways for users to chain LLM steps together while controlling what inference gets exhibited without the infrastructure concerns or data risks typically associated with external cloud services, presents a unique opportunity to democratize AI capabilities. By ensuring that users can interact with and execute complex AI
```

Comment:
```suggestion
Self-hosted, privately managed model deployments hit on many of the privacy and security modules taught in our course. Incorporating ways for users to chain LLM steps together while controlling what inference gets exhibited without the infrastructure concerns or data risks typically associated with external cloud services, presents a unique opportunity to make AI capabilities broadly available. By ensuring that users can interact with and execute complex AI
```

Totally optional edit! IMO "democratize" is an abused term when what we really mean is making something more available.